xapi-compression/xapi_compression.ml

```diff
@@ -73,10 +77,14 @@ module Make (Algorithm : ALGORITHM) = struct
       (fun () ->
         let failwith_error s =
           let mode =
-            if mode = Compress then "Compression" else "Decompression"
+            if mode = Compress then "Compress" else "Decompress"
```

Comment:
I think it's fine as is but why are Compress and Decompress capitalized in this context? For an algebraic type like the mode I feel a `match` would be slightly more idiomatic.